test/optional-validation.test.js

```diff
@@ -0,0 +1,504 @@
+// This test written in mocha+should.js
+var should = require('./init.js');
+var async = require('async');
+var db, User, options, filter;
+var j = require('../');
+var ValidationError = j.ValidationError;
+
+describe('optional-validation', function () {
+
+  before(function (done) {
+    db = getSchema();
+    User = db.define('User', {
+      seq: {type: Number, index: true},
+      name: {type: String, index: true, sort: true},
+      email: {type: String, index: true},
+      birthday: {type: Date, index: true},
+      role: {type: String, index: true},
+      order: {type: Number, index: true, sort: true},
+      vip: {type: Boolean}
+    });
+
+    options = {};
+    filter = {fields: ['name', 'id']};
+
+    db.automigrate(['User'], done);
+
+  });
+
+  beforeEach(function (done) {
+    User.destroyAll(function () {
+      delete User.validations;
+      done();
+    });
+  });
+
+  describe('no model setting', function () {
+
+    it('should throw on create with validate:true with invalid data', function (done) {
+      var options = {validate: true};
+      User.validatesPresenceOf('name');
+
+      User.create({name: null}, options, function (e, d) {
+        should.exist(e);
+        e.should.be.instanceOf(Error);
+        e.should.be.instanceOf(ValidationError);
+        done();
+      });
+    });
+
+    it('should NOT throw on create with validate:false with invalid data', function (done) {
+      var options = {validate: false};
+      User.validatesPresenceOf('name');
+
+      User.create({name: null}, options, function (e, d) {
+        should.not.exist(e);
```

Comment:
Please use `if (err) return done(err)` - it gives more helpful stack traces on error. Also please apply the same changes to other places where you call `should.not.exist(e)`.